src/components/Input.react.js

```diff
@@ -41,6 +41,20 @@ export default class Input extends Component {
                     if (fireEvent) {
                         fireEvent({event: 'blur'});
                     }
+                    if (setProps) {
```

Comment:
A small improvement here could be that if `setProps` is set, the component doesn't need to keep state itself, i.e. if `setProps` is set you don't need to fire `setState`. If you were to implement that, you'd also need to add that logic to `componentWillReceiveProps`. We had a small discussion about that in #331 - it won't cause any bugs like that here but it's still a small gain to not keep unnecessary state in this component. Up to you if you want to add that now, we could always add it later if you want to just merge this first.

Reply:
But it's not using the state.

Reply:
Ah I was looking at line 29. That's something for another time then, feel free to merge!

Reply:
Is there a note about that in our react dev guide ? In the SuggestionsInput, I was using `componentWillReceiveProps` to setState on the value, but also setState before calling setProps on the value and the value in the props and state was always different so it lead to a bug where no suggestions beyond the first was registered so it a very valid concern to think about when developing dash component.

Reply:
Exactly, like I mentioned above I found a bug with that recently too in #331. In the Input component I don't think it'll lead to any bugs though. But it's not in the React for Python devs guide, no. Would be good to add that!